src/main/java/uk/gov/hmcts/cmc/claimstore/services/JobSchedulerService.java

```diff
@@ -0,0 +1,123 @@
+package uk.gov.hmcts.cmc.claimstore.services;
+
+import com.google.common.collect.ImmutableMap;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.cmc.claimstore.idam.models.User;
+import uk.gov.hmcts.cmc.claimstore.jobs.NotificationEmailJob;
+import uk.gov.hmcts.cmc.domain.models.Claim;
+import uk.gov.hmcts.cmc.scheduler.model.JobData;
+import uk.gov.hmcts.cmc.scheduler.services.JobService;
+
+import java.time.LocalDate;
+import java.time.ZoneOffset;
+import java.util.Map;
+
+@Service
+public class JobSchedulerService {
+
+    private final JobService jobService;
+    private final UserService userService;
+    private final int firstReminderDay;
+    private final int lastReminderDay;
+
+    public JobSchedulerService(
+        JobService jobService,
+        UserService userService,
+        @Value("${dateCalculations.firstResponseReminderDay}") int firstReminderDay,
+        @Value("${dateCalculations.lastResponseReminderDay}") int lastReminderDay
+    ) {
+        this.jobService = jobService;
+        this.userService = userService;
+        this.firstReminderDay = firstReminderDay;
+        this.lastReminderDay = lastReminderDay;
+    }
+
+    public void scheduleEmailNotificationsForDefendantResponse(
+        String authorisation,
+        Claim claim
+    ) {
+        LocalDate responseDeadline = claim.getResponseDeadline();
+
+        Map<String, Object> data = getNotificationData(authorisation, claim, responseDeadline);
+
+        jobService.scheduleJob(
+            getFirstReminderJobData(claim, data),
+            responseDeadline.minusDays(firstReminderDay).atStartOfDay(ZoneOffset.UTC)
+        );
+
+        jobService.scheduleJob(
+            getLastReminderJobData(claim, data),
+            responseDeadline.minusDays(lastReminderDay).atStartOfDay(ZoneOffset.UTC)
+        );
+
+    }
+
+    public void rescheduleEmailNotificationsForDefendantResponse(
+        String authorisation,
+        Claim claim,
+        LocalDate responseDeadline
+    ) {
+        Map<String, Object> data = getNotificationData(authorisation, claim, responseDeadline);
+
+        jobService.rescheduleJob(
+            getFirstReminderJobData(claim, data),
+            responseDeadline.minusDays(firstReminderDay).atStartOfDay(ZoneOffset.UTC)
+        );
+
+        jobService.rescheduleJob(
+            getLastReminderJobData(claim, data),
+            responseDeadline.minusDays(lastReminderDay).atStartOfDay(ZoneOffset.UTC)
+        );
+
+    }
+
+    private JobData getLastReminderJobData(Claim claim, Map<String, Object> data) {
```

Comment:
There is a lot of duplicates, please refactor it have one method `createReminderJobData(Claim claim, int numberOfDaysBeforeDeadline)`